### ISSUE TITLE
Can't list servers when one is booted from volume

### ISSUE THREAD

**Issue by jccomputing** [gophercloud/gophercloud#11]
When I list server instances, it fails when an instance is booted from volume with the following error: `json: cannot unmarshal string into Go value of type map[string]interface {}`

This code will trigger the bug, provided you have proper environment variables and at least one volume booted server:
```go
package main

import (
	"log"
	"os"

	"github.com/gophercloud/gophercloud"
	"github.com/gophercloud/gophercloud/openstack"
	"github.com/gophercloud/gophercloud/openstack/compute/v2/servers"
)

func main() {
	opts, _ := openstack.AuthOptionsFromEnv()
	provider, _ := openstack.AuthenticatedClient(opts)
	client, _ := openstack.NewComputeV2(provider, gophercloud.EndpointOpts{Region: os.Getenv("OS_REGION_NAME")})
	listopts := servers.ListOpts{}
	pager := servers.List(client, listopts)
	_, err := pager.AllPages()
	if err != nil {
		log.Fatalln(err)
	}

}
```

The JSON returned in that case contains: `image: ""` and [ExtractInto](https://github.com/gophercloud/gophercloud/blob/master/results.go#L58) tries to convert this empty JSON string to a [map[string]interface{}](https://github.com/gophercloud/gophercloud/blob/master/openstack/compute/v2/servers/results.go#L166) which fails.

I reproduced it on a Juno and a Liberty release.

**Comment by jrperritt** [comment]
Ack. Similar to #7 . Will fix.

**Comment by jrperritt** [comment]
closed by #18
